File: BowlingGame/#7Kata/BowlingGame/src/main/java/Game.java

public class Game {

	private int[] rolls = new int[21];
	private int current = 0;

	public void roll(int pins) {
		rolls[current++] = pins;
	}

	public int score() {
		int score = 0;
		int frameIndex = 0;
		for (int frame = 0; frame < 10; frame++) {
			if (isStrike(frameIndex)) {
				score += strikeScore(frameIndex);
				frameIndex++;
			} else if (isSpare(frameIndex)) {
				score += spareScore(frameIndex);
				frameIndex += 2;
			} else {
				score += rolls[frameIndex] + rolls[frameIndex + 1];
				frameIndex += 2;
			}
		}
		return score;
	}

	private int spareScore(int frameIndex) {
		return 10 + rolls[frameIndex + 2];
	}

	private int strikeScore(int frameIndex) {
		return 10 + rolls[frameIndex + 1] + rolls[frameIndex + 2];
	}

	private boolean isSpare(int frameIndex) {
		return rolls[frameIndex] + rolls[frameIndex + 1] == 10;
	}

	private boolean isStrike(int frameIndex) {
		return rolls[frameIndex] == 10;
	}
}

File: BowlingGame/#8Kata/BowlingGame.Model/Game.cs
namespace BowlingGame.Model
{
    public class Game
    {
        private int[] Rolls = new int[21];
        private int Current = 0;

        public int Score { get; set; }
        public void roll(int pins)
        {
            Rolls[Current++] += pins;
        }

        public int score()
        {
            int result = 0;
            int frameIndex = 0;

            for (int frame = 0; frame < 10; frame++)
            {
                if (IsStrike(frameIndex))
                {
                    result += StrikeScore(frameIndex);
                    frameIndex++;
                }
                else if (IsSpare(frameIndex))
                {
                    result += SpareScore(frameIndex);
                    frameIndex += 2;
                }
                else
                {
                    result += CommonScore(frameIndex);
                    frameIndex += 2;
                }
            }

            return result;
        }

        private int CommonScore(int frameIndex)
        {
            return Rolls[frameIndex] + Rolls[frameIndex + 1];
        }

        private int SpareScore(int frameIndex)
        {
            return 10 + Rolls[frameIndex + 2];
        }

        private int StrikeScore(int frameIndex)
        {
            return 10 + Rolls[frameIndex + 1] + Rolls[frameIndex + 2];
        }

        private bool IsSpare(int frameIndex)
        {
            return Rolls[frameIndex] + Rolls[frameIndex + 1] == 10;
        }

        private bool IsStrike(int frameIndex)
        {
            return Rolls[frameIndex] == 10;
        }
    }
}
